File: python/dispatcher.py
from handlers import Handlers
from command import Command

class CommandDispatcher:
    def __init__(self):
        self.handlers = {
            "PING": Handlers.ping,
            "STATUS": Handlers.status,
        }

    def dispatch(self, command):
        handler = self.handlers.get(command.name)
        if handler is None:
            return "UNKNOWN_COMMAND"
        return handler()

File: java/src/main/java/Dispatcher.java
import java.util.Map;
import java.util.function.Supplier;

public class Dispatcher {

    private final Map<String, Supplier<String>> handlers;

    public Dispatcher(Map<String, Supplier<String>> handlers) {
        this.handlers = handlers;
    }

    public String dispatch(Command command) {
        Supplier<String> handler = handlers.get(command.getName());
        if (handler == null) {
            return "UNKNOWN_COMMAND";
        }
        return handler.get();
    }
}
